gateway-service/src/main/java/org/zowe/apiml/gateway/security/config/NewSecurityConfiguration.java

```diff
@@ -0,0 +1,394 @@
+                HttpMethod.GET,
+                false,
+                authenticationManager);
+        }
+    }
+
+    /**
+     * Query and Ticket endpoints share single filter that handles auth with and without certificate. This logic is encapsulated in the queryFilter or ticketFilter.
+     * Ticket endpoint does require certificate to be present in RequestContext
+     */
+
+    @Configuration
+    @RequiredArgsConstructor
+    @Order(3)
+    class Ticket extends WebSecurityConfigurerAdapter {
+
+        private final AuthenticationProvider tokenAuthenticationProvider;
+
+        @Override
+        protected void configure(AuthenticationManagerBuilder auth) {
+            auth.authenticationProvider(tokenAuthenticationProvider); // for authenticating Tokens
+        }
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            baseConfigure(http.requestMatchers().antMatchers(
+                authConfigurationProperties.getGatewayTicketEndpoint(),
+                authConfigurationProperties.getGatewayTicketEndpointOldFormat()
+            ).and()).authorizeRequests()
+                .anyRequest().authenticated()
+                .and()
+                .logout().disable() // logout filter in this chain not needed
+                .x509() //default x509 filter, authenticates trusted cert, ticketFilter(..) depends on this
+                .subjectPrincipalRegex(EXTRACT_USER_PRINCIPAL_FROM_COMMON_NAME)
+                .userDetailsService(new SimpleUserDetailService())
+                .and()
+                .addFilterBefore(ticketFilter("/**", authenticationManager()), UsernamePasswordAuthenticationFilter.class);
+        }
+
+        private QueryFilter ticketFilter(String ticketEndpoint, AuthenticationManager authenticationManager) {
+            return new QueryFilter(
+                ticketEndpoint,
+                successfulTicketHandler,
+                handlerInitializer.getAuthenticationFailureHandler(),
+                authenticationService,
+                HttpMethod.POST,
+                true,
+                authenticationManager);
+        }
+    }
+
+    @Configuration
+    @RequiredArgsConstructor
+    @Order(4)
+    class CertificateProtectedEndpoints extends WebSecurityConfigurerAdapter {
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            baseConfigure(http.requestMatchers()
+                    .antMatchers(HttpMethod.DELETE,CacheServiceController.CONTROLLER_PATH + "/**")
+                    .antMatchers(AuthController.CONTROLLER_PATH + AuthController.INVALIDATE_PATH, AuthController.CONTROLLER_PATH + AuthController.DISTRIBUTE_PATH).and()
+                ).authorizeRequests()
+                .anyRequest().authenticated()
+                .and()
+                .logout().disable() // logout filter in this chain not needed
+                .x509() // default x509 filter, authenticates trusted cert

```

Comment:
Does it work with routing from outside via gateway? Don't we need to change this somehow?

Reply:
It looks as if GW certificate will always be used

Reply:
Calling things through Gateway with certificate is a topic that is encapsulated in `HttpClientChooser` class. It's not related to gateway's filterchain. Do you have specific concern in mind? 
If you look at the ant matchers for this filter chain, you will see that they are internal gateway endpoints used by other gateway instances called directly against each other.

Reply:
I know, and if what we were asking with Andrej is the case, then it was probably an issue before as well. And I understand how these are sued typically and the typical case is totally fine. 

The concern is could we call these endpoint via gateway like for other onboarded services? And if we do so, would it allow us to access them with less privileges?